exporter/elasticsearchexporter/model.go

```diff
@@ -154,67 +176,12 @@ func (m *encodeModel) encodeLogOTelMode(resource pcommon.Resource, resourceSchem
 	// updated by the router.
 	// Move them to the top of the document and remove them from the record
 	attributeMap := record.Attributes()
-
-	forEachDataStreamKey := func(fn func(key string)) {
-		for _, key := range datastreamKeys {
-			fn(key)
-		}
-	}
-
-	forEachDataStreamKey(func(key string) {
-		if value, exists := attributeMap.Get(key); exists {
-			document.AddAttribute(key, value)
-			attributeMap.Remove(key)
-		}
-	})
-
+	addDataStreamAttributes(&document, attributeMap)
+	stripDataStreamAttributes(attributeMap)
```

Comment:
`pcommon.Map#Get` could get expensive with more attributes though I don't think we would have a huge number of keys but it is also not helping with readability of the code IMO. Also I am not a fan of these util functions. How about:

```go
	attributeMap.RemoveIf(func(k string, v pcommon.Value) bool {
		// just a few keys in datastream keys
		if slices.Contains(datastreamKeys, k) {
			// Add attribute will not add empty values
			document.AddAttribute(k, v)
			return true
		}
		return false
	})
```

Reply:
I agree that it is not great. It was designed as such as we need to call `addDataStreamAttributes` for record level attributes, and `stripDataStreamAttributes` for all resource, scope, and record level attributes. Splitting them into 2 functions makes it easier to understand. Performance would have been fine if attributes Get was O(1), but the current implementation isn't.

Would you still prefer implementing it inline for all 3 kinds of attributes?

Reply:
> Would you still prefer implementing it inline for all 3 kinds of attributes?

I think the function call is simple enough to inline and would also be better for readability. So, my vote is to inline.

Reply:
done, looking much cleaner now.